echopype/convert/set_groups_ad2cp.py

```diff
@@ -254,54 +246,166 @@ def set_platform(self) -> xr.Dataset:
         )
         return set_encodings(ds)
 
-    def set_beam(self) -> xr.Dataset:
+    def set_beam(self) -> List[xr.Dataset]:
         # TODO: should we divide beam into burst/average (e.g., beam_burst, beam_average)
         # like was done for range_bin (we have range_bin_burst, range_bin_average,
         # and range_bin_echosounder)?
-        ds = self._make_dataset(
-            {
-                "num_beams": "number_of_beams",
-                "coordinate_system": "coordinate_system",
-                "num_cells": "number_of_cells",
-                "blanking": "blanking",
-                "cell_size": "cell_size",
-                "velocity_range": "velocity_range",
-                "echosounder_frequency": "echosounder_frequency",
-                "ambiguity_velocity": "ambiguity_velocity",
-                "dataset_description": "data_set_description",
-                "transmit_energy": "transmit_energy",
-                "velocity_scaling": "velocity_scaling",
-                "velocity_data_burst": "velocity_burst",
-                "velocity_data_average": "velocity_average",
-                "amplitude_data_burst": "amplitude_burst",
-                "amplitude_data_average": "amplitude_average",
-                "correlation_data_burst": "correlation_burst",
-                "correlation_data_average": "correlation_average",
-                "correlation_data_echosounder": "correlation_echosounder",
-                "echosounder_data": "amplitude_echosounder",
-                "figure_of_merit_data": "figure_of_merit",
-                "altimeter_distance": "altimeter_distance",
-                "altimeter_quality": "altimeter_quality",
-                "ast_distance": "ast_distance",
-                "ast_quality": "ast_quality",
-                "ast_offset_100us": "ast_offset_100us",
-                "ast_pressure": "ast_pressure",
-                "altimeter_spare": "altimeter_spare",
-                "altimeter_raw_data_num_samples": "altimeter_raw_data_num_samples",
-                "altimeter_raw_data_sample_distance": "altimeter_raw_data_sample_distance",
-                "altimeter_raw_data_samples": "altimeter_raw_data_samples",
-            }
-        )
-        ds = ds.assign_attrs({"pulse_compressed": self.pulse_compressed})
+        beam_groups = []
+        self._beamgroups = []
+
+        # burst
+        for packet in self.parser_obj.packets:
```

Comment:
This appears as a for loop but it doesn't actually loop through all packets and the looping through packets actually happens in `_make_dataset`. I wonder if it would be more explicit if you do a smaller loop to check if certain type of packets exist in the file (since the packet types to consider is finite), and then from that use the same `beam_groups.append` and `self._beamgroups.append` sequence to assemble the datasets and group sequence/descriptions.

Also, please change the sequence so that Average is the first before Burst, so the sequence of beam groups (whenever they exist) is: Average - Burst - Echosounder - Echosounder_raw.